npm/git-cliff/package.json

```diff
@@ -22,6 +22,9 @@
       }
     }
   },
+  "files": [
+    "lib"
```

Comment:
Could you include the source Typescript files as well?

Reply:
There is absolutely no reason to do so. Types are provided through the output `.d.ts` files and node, bun and deno all will never even load the source typescript files.

Reply:
Also due to the export mapping (a few lines above this) the files in src won't even be accessible. Node will just say they're not exported. The only way to read then would with a manual readFile call.

Reply:
Well, my build system Parcel is capable of using the source files directly. I also prefer untransformed source code that is on GitHub. Given what has been going on with the xz distribution, you should consider this.

Reply:
I don't see any relationship to what happened to xz. It's also not "going on", it's already resolved. Please show me the documentation where parcel says they use the source files. I've used parcel before and it doesn't. Please also explain what your exact use case is that you use parcel to not only consume the git-cliff npm package but also parcel bundle it. Parcel is for bundling webapps after all.

Reply:
Here's the relevant Parcel documentation:
https://parceljs.org/features/dependency-resolution/#package-entries

I am not sure why this is controversial. 
The solution is simple, instead of only distributing transpiled code, the Typescript source can be included as well.

Reply:
Read the documentation carefully. The package.json is consulted.

- source -> this package doesn't define it, also this only affects symlinked packages which this wouldn't be
- exports -> this package defines this, directing to lib

The rest becomes irrelevant because an entrypoint has now been found, but to indulge the others:

- browser -> not defined
- module -> goes to lib
- main -> goes to lib

As for why it's controversial, it unnecessarily bloats the tarball causing the download size to be larger. 

As for JSR, this package does not (yet) get published to jsr so that's a non-issue and for later. Furthermore jsr is still in its infancy and I'd first like to see larger adoption. It's also a registry made by the Deno team whose Node compat layer is infamously horrible (especially when compared to Bun) so I have my opinionated reservations about whether it will even be relevant at all a year from now.

Reply:
I'm not well versed in these topics so I cannot comment. But I would appreciate a ELI5 summary 😊

Reply:
Sure! 

@aminya wants me to include the TypeScript files from `src` in the tarball. This is something that no library ever does. Search the favourite big-name packages written in TypeScript on https://yarnpkg.com as they have a file viewer for what's in the tarball, and you'll see that none of them include the source TypeScript files. This is because runtimes and bundlers will never access them. This is despite what @aminya says about Parcel, as described by my previous comment. Even if Parcel *would* access those files, it would be the only bundler to do so and it's far from the [most popular bundler (npmtrends)](https://npmtrends.com/esbuild-vs-parcel-vs-rollup-vs-snowpack-vs-vite-vs-webpack). Ultimately, including the source TypeScript files is unnecessary bloat and increasing filesize for no actual gain.

Reply:
Optimizing the package size seems unnecessary, and the fact that the popular packages do not provide source files just reveals the wide attack surface in open source.
 
 The question remains though. How can I make sure the JavaScript files in the dist are exactly derived from the source TypeScript files in the repository? Do you have a byte-for-byte reproducible build and release process?

Reply:
Thanks for the detailed summary and your thoughts! I agree that we should still have faith in the open source community. I don't have a strong preference between including/not including the TS files due to the fact that it's an whole another ecosystem that I'm not a part of. I can still very clearly see @aminya's concerns, but I'd say we can discuss this elsewhere for the sake of not blocking this PR.

I'm happy to include the TS files if there are strong arguments with practical and proven reproducibility along with a PR.